kotlin/custom-dictionary-style/src/main/java/com/esri/arcgisruntime/sample/customdictionarystyle/MainActivity.kt

```diff
@@ -0,0 +1,57 @@
+package com.esri.arcgisruntime.sample.customdictionarystyle
+
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import com.esri.arcgisruntime.data.ServiceFeatureTable
+import com.esri.arcgisruntime.layers.FeatureLayer
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.Basemap
+import com.esri.arcgisruntime.mapping.Viewpoint
+import com.esri.arcgisruntime.symbology.DictionaryRenderer
+import com.esri.arcgisruntime.symbology.DictionarySymbolStyle
+import kotlinx.android.synthetic.main.activity_main.*
+
+class MainActivity : AppCompatActivity() {
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    // create a feature layer from a service feature table
+    val featureTable = ServiceFeatureTable(getString(R.string.restaurants_url))
+    val featureLayer = FeatureLayer(featureTable).apply {
+      // use a custom style to create a dictionary renderer and set it to the feature layer renderer
+      renderer = DictionaryRenderer(
+        DictionarySymbolStyle.createFromFile(
+          getExternalFilesDir(null)?.path + getString(R.string.restaurant_stylx_path)
+        )
+      )
+    }
+
+    // set the map view's viewpoint to the feature layer extent when loaded
+    featureLayer.addDoneLoadingListener {
```

Comment:
you could add this to the `FeatureLayer(featureTable).apply { ...}` block above

Reply:
That'd be my intuition too, but I'm trying to keep to the apply block as a sort of extended constructor as per @tschie's comment above.